scripts/corelli/calibration/bank.py

```diff
@@ -180,10 +186,83 @@ def collect_bank_fit_results(output_workspace: str,
     return workspace
 
 
-def criterium_peak_pixel_position(peak_table: InputTable,
+def criterion_peak_vertical_position(peak_table: InputTable,
+                                     summary: Optional[str] = None,
+                                     zscore_threshold: float = 2.5,
+                                     deviation_threshold: float = 0.0035) -> np.ndarray:
+    r"""
+    Flag tubes whose wire shadows vertical positions deviate considerably from the vertical positions when
+    averaged for all tubes in the bank.
+
+    .. math::
+
+      <p_i> = \frac{1}{n_t} \Sum_{j=1}^{n_t} p_{ij}
+      \delta_j^2 = \frac{1}{n_w} \Sum (p_{ij} - <p_i>)^2
+      assert d_j < threshold
+
+    :param peak_table: vertical positions of the wire shadows for each tube, in meters
+    :param summary: name of output Workspace2D containing deviations and Z-score for each tube.
+    :param zscore_threshold: maximum Z-score for the vertical positions of a tube.
+    :param deviation_threshold: maximum deviation (in meters) for the vertical positions of the wire shadows.
+    :return: array of booleans, one per tube. `True` is the tube passes the acceptance criterion, `False` otherwise.
+    """
+    table = mtd[str(peak_table)]  # handle to the peak table
+    peak_count = table.columnCount() - 1  # the first column contains the names of the tubes
+    # `positions_average` stores the vertical position for each wire shadow, averaged for all tubes
+    positions_average = [np.mean(table.column(column_number)) for column_number in range(1, 1 + peak_count)]
+
+    deviations = list()  # a measure of how much the vertical positions deviate from the mean positions
```

Comment:
just bikeshedding with typing: `deviation:List[ np.ndarray ] = []`  is a bit more clear